nncf/experimental/post_training/statistics/aggregator.py

```diff
@@ -44,7 +43,9 @@ def collect_statistics(self, model: ModelType) -> None:
         Collects statistics for registered StatisticPoints.
         The statistics are stored in self.statistic_points.
         """
-        self.engine.set_model(self._prepare_model_for_statistics_collection(model))
+        model_transformer = MODEL_TRANSFORMERS.get()
```

Comment:
If you are setting up future groundwork for dispatching correct transformer for a correct framework, then I'm not sure that this is the right place. I don't know where the correct place should be, though. Perhaps the `__init__` function of this same class should be more appropriate, with a dispatch based on the specific `Engine`, or perhaps `__init__` should already receive the correct object that is a subclass of `StaticModelTransformerBase`.